keps/sig-node/3675-resource-plugin-manager/README.md

```diff
@@ -0,0 +1,1229 @@
+back to user and the Pod allocation is cancelled.
+
+`cciAddContainerResource` function is called before container start with a given 
+Pod name and container name, cgroup of Pod and the container id. The driver than
+performs an allocation of the admitted resource-set.  In case of failure of the
+driver and error is returned.
+
+`cciRemoveContainerResource` function is called to free the cpu resources taken
+by a container. The functions returns nil if the operation was successful or an
+error if the operation failed on the plugin side.
+
+    /* 
+    CCIDriver a grpc service interface for cci resour ubelet d ent drivers.
+    The interface provides admission, allocation and cleanup entrypoints.
+    Drivers are registered  ubeletlet plugins and will be called by the 
+    cci resource manager for Pods which are associated with this driver.
+    */
+    service CCIDriver {
+      //cciAdmit admission function to reserve resources for a given container
+      rpc cciAdmit(AdmitRequest) returns (AdmitResponse) {}
+  
+      //cciAddContainerResource allocation entry point for admitted containers
+      rpc cciAddContainerResource (AddResourceRequest)
+        returns (AddResourceResponse) {}
+
+      //cciRemoveContainerResource clea ubelet dted resources for a given container
+      rpc cciRemoveContainerResource (RemoveResourceRequest)
+        returns (RemoveResourceResponse) {}
+    }
+
+5.	CCI Drivers Factory API<br>
+The KEP includes a new staged API which enabled the CCI driver creation. The 
+API can be used by a driver implementor to start a driver and automatically
+register it against Kubelet's CCI Resource Manager.
+
+6.	CCI Drivers Registration<br>
+The CCI driver registration can be handled by t ubeletlet plugin framework. 
+This approach is already used with device plugins and DRA plugins. The approach
+will be sufficient the cover plugin registration, status and health-check functionality.
+
+##### Allocation & Container Removal Flow
+
+Admission and adding a Container:<br>
+![image](CCIAllocationSequenceDiagram.jpg)<br>
+Fig. 3.: Sequence of container allocation which uses CCI driver<br><br>
+
+
+The lifetime events are triggered by the container manager and internal lifecycle manager in 
+exact order for admitting, adding, and removing containers. As shown on Fig . 3  an admit 
+container is invoked for containers inside Pods requiring a resource driver before calling 
+the admit on the cpu manager side. On success the resource set result is added to the CCI 
+store. The admit call on cpu manager side will trigger the CCI policy which will execute a 
+lookup inside the store to get the assigned cpuset for the new container.  If the operation 
+fails an error will be reported back to the user. On success the data will be stored in the 
+cpu manager state which then gets accessed by add container call. All blocking rpc calls are 
+configured in alpha with a reasonable timeout. 
```

Comment:
Can you also describe when `cciAddContainerResource` is called? I'm thinking if we actually need a separate admission call.

Reply:
added some more text to clarify that, we distinguish between admission and allocation stage similar to CPU manager.